File: java_code/warmup2/last2.java
package java_code.warmup2;

//codingbat link: https://codingbat.com/prob/p178318

public class last2 {

    public int last2(String str) {
        if (str.length() < 2)
            return 0;

        String ending = str.substring(str.length() - 2);
        int count = 0;

        for (int i = 0; i < str.length() - 2; i++) {
            if (str.substring(i, i + 2).equals(ending)) {
                count++;
            }
        }

        return count;
    }
}

File: python/warmup2/last2.py
#codingbat link: https://codingbat.com/prob/p145834


def last2(str):

    if (len(str) < 2):
        return 0

    else:
        lastStr = str[len(str) - 2:]
        count = 0
        for i in range(len(str) - 2):
            if (str[i:i + 2] == lastStr):
                count = count + 1

        return count
